flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceBounded.java

```diff
@@ -129,6 +129,9 @@ protected List<Row> run(
 
     try (CloseableIterator<Row> iter = stream.executeAndCollect()) {
       return Lists.newArrayList(iter);
+    } catch (Exception e) {
+      // To retrieve the underlying exception information that actually caused the task failure.
+      throw (RuntimeException) e.getCause().getCause().getCause().getCause().getCause().getCause();
```

Comment:
why it is important to extract the root cause exception? also this chaining is very fragile. if we just want the root cause, there are util method to do that.

Reply:
Yes, it has been corrected.